### PR TITLE
Upgraded Borg Snack Dispenser

### PR DESCRIPTION
**Pull request by Runian**
# Document the changes in your pull request
This ports Skyrat-SS13/Skyrat-tg/pull/11941 and mostly copies the features of the treat fabricator and cookie synth.

Cyborg modules that have some form of snack dispenser now have a newer version of it:
- Peacekeeper gets one that only gives cookies.
- Medical, Service, and Clown gets one that gives lollipops, gumballs, and vanilla icecream.

When emagged, all thrown snacks deal 3 force damage and all cookies have 10u of Chloral Hydrate.
Alt-click to toggle between dispensing and throwing.
Costs 50 power per use.
Locked behind Cyborg Upgrades: Utility.

There is a new upgrade that allows them to give snacks that they don't normally have to a total of 6 snacks: bacon, cloth cookies, normal cookies, lollipops, gumballs, and vanilla ice cream.

This upgrade can be printed for 1-sheet worth of glass and 1-sheet worth of metal at an exosuit fabricator.

A minor bugfix in which `datum/thrownthing/finalize()` didn't signal `COMSIG_MOVABLE_THROW_LANDED`.

Throwing:
throwtime

Placing on tables, floor, and to others:
placetime

# Spriting
"odd cookie" (left) and "strip of bacon" (right)
image

# Wiki Documentation
Add this to [Cyborg Upgrades](https://wiki.yogstation.net/wiki/Guide_to_robotics#Cyborg_Upgrades) at guide to robotics.
# Changelog

:cl:  
rscadd: Borg Snack Dispenser has been added.
rscadd: A new cyborg upgrade can be printed to give Cyborgs with a Borg Snack Dispenser access to more snacks.
rscdel: The cyborg treat fabricator has been removed.
tweak: Peacekeeper Cyborgs starts with a Borg Snack Dispenser that gives only cookies.
tweak: Medical, Service, and Clowns Cyborgs starts with a Borg Snack Dispenser that gives lollipops, gumballs, and vanilla ice cream.
bugfix: The finalizing of a thrown item properly calls COMSIG_MOVABLE_THROW_LANDED signal.
/:cl:

### CODE REVIEW

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -553,6 +553,99 @@
 	head.color = color2
 	add_overlay(head)
 
+/obj/item/borg_snack_dispenser
+	name = "\improper Automated Borg Snack Dispenser"
+	desc = "Has the ability to automatically print many differnt forms of snacks. Now Lizard approved!"
```
Comment (ToasterBiome) [inline code/game/objects/items/robot/robot_items.dm]: differnt

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
spelling error + add a cooldown to this. Cant wait for 50,000 items on the ground to be annoying.

File: code/game/objects/items/robot/robot_items.dm (hunk 1 above)
Comment (Runian) [reply code/game/objects/items/robot/robot_items.dm]: Donezo.

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
use actual cooldowns, look for COOLDOWN_DECLARE to see how it's used

**Review by ToasterBiome — APPROVED** [review]

**Review by ynot01 — CHANGES_REQUESTED** [review]
Remove the old treat fabricator (/obj/item/borg/lollipop, /obj/item/rsf/cookiesynth) and replace it with this one but limiting options for snacks (by proc or by subtype). Also add gumballs. No sense in having 3 different types of snack dispensers.

**Review by ynot01 — APPROVED** [review]

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -439,9 +443,21 @@
...
+	// Vanilla Icecream & Setting 'snack.name' Early
+	if(istype(snack, /obj/item/reagent_containers/food/snacks/icecream))
+		var/obj/item/reagent_containers/food/snacks/icecream/icecream = snack
+		icecream.add_ice_cream("vanilla")
+		icecream.desc = "Eat the ice cream."
+		
 	to_chat(patron, span_notice("[user] dispenses a [snack.name] into your empty hand and you reflexively grasp it."))
 	to_chat(user, span_notice("You dispense a [snack.name] into the hand of [patron]."))
 
+	if(snack && istype(snack, /obj/item/reagent_containers/food/snacks/icecream))
```
Comment (ynot01) [inline code/game/objects/items/robot/robot_items.dm]: why twice?

Comment (Runian) [reply code/game/objects/items/robot/robot_items.dm]: Forgot to remove.

**Review by ynot01 — APPROVED** [review]

**Pull request merged**